src/DotVVM.Framework/Controls/Infrastructure/BodyResourceLinks.cs

```diff
@@ -56,7 +56,7 @@ internal static string RenderWarnings(IDotvvmRequestContext context)
             foreach (var w in collector.GetWarnings())
             {
                 var msg = JsonConvert.ToString(w.ToString(), '"', StringEscapeHandling.EscapeHtml);
-                result += $"dotvvm.log.logWarning({msg});\n";
+                result += $"console.warn({msg});\n";
```

Comment:
We should be using `dotvvm.log` wrappers

Reply:
The problem is that this code runs before dotvvm is initialized.